java/src/test/java/org/rocksdb/RocksDBTest.java

```diff
@@ -1423,6 +1423,33 @@ public void getLiveFilesMetadataWithChecksum() throws RocksDBException {
     }
   }
 
+  @Test
+  public void getColumnFamilyMetadataWithChecksum() throws RocksDBException {
+    final Properties props = new Properties();
+    props.put("file_checksum_gen_factory", "FileChecksumGenCrc32cFactory");
+    final String dbPath = dbFolder.getRoot().getAbsolutePath();
+
+    try (final DBOptions dbOptions = DBOptions.getDBOptionsFromProps(props);
+         final ColumnFamilyOptions cfOptions = new ColumnFamilyOptions();
+         final Options options = new Options(dbOptions, cfOptions).setCreateIfMissing(true)) {
+      try (final RocksDB db = RocksDB.open(options, dbPath);
+           final WriteOptions writeOptions = new WriteOptions().setDisableWAL(true)) {
+        db.put("key".getBytes(UTF_8), "value".getBytes(UTF_8));
+      }
+
+      try (final RocksDB db = RocksDB.open(options, dbFolder.getRoot().getAbsolutePath())) {
+        ColumnFamilyMetaData metadata = db.getColumnFamilyMetaData(); // Exception here
```

Comment:
In the reproduction that I had, it was only the _second_ call to `getColumnFamilyMetaData` that would throw an exception. We might consider testing that case as well. But does this code _also_ fail on `8.11.fb` and `9.0.fb`?

Reply:
> But does this code also fail on 8.11.fb and 9.0.fb

You would need to test it. We are unlikely to backport this directly

Reply:
Hello @neilramaswamy,

I was able to get the exception on first call. Didn't need the second one. As I mentioned, what is importand is the SST file on disk. This was the _trigger_ for exception.

Radek

Reply:
> You would need to test it. We are unlikely to backport this directly

Yeah, I was more curious, wasn't expecting a backport. But I did verify that on 8.11.fb and 9.0.fb this test indeed fails. Thanks Radek!

I'm curious though, @adamretter, is this _fix_ something that you would backport? I imagine that breaking `getColumnFamilyMetaData` calls for an SST file on disk could impact a lot of users.